sdk/maps/Azure.Maps.Search/README.md

```diff
@@ -0,0 +1,185 @@
+
+We also need **Azure Maps Client ID** which can get from Azure Maps page > Authentication tab > "Client ID" in Azure Active Directory Authentication section.
+
+```C# Snippet:InstantiateSearchClientViaAAD
+// Create a MapsSearchClient that will authenticate through AAD
+var credential = new DefaultAzureCredential();
+var clientId = "<My Map Account Client Id>";
+MapsSearchClient client = new MapsSearchClient(credential, clientId);
+```
+
+## Key concepts
+
+`MapsSearchClient` is designed for:
+
+* Communicate with Azure Maps endpoint to query addresses or points of locations
+* Communicate with Azure Maps endpoint to request the geometry data such as a city or country outline for a set of entities
+* Communicate with Azure Maps endpoint to perform a free form search inside a single geometry or many of them
+
+Learn more about examples in [samples](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/maps/Azure.Maps.Search/tests/Samples)
+
+### Thread safety
+
+We guarantee that all client instance methods are thread-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-methods-thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across threads.
+
+### Additional concepts
+<!-- CLIENT COMMON BAR -->
+[Client options](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#configuring-service-clients-using-clientoptions) |
+[Accessing the response](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#accessing-http-response-details-using-responset) |
+[Long-running operations](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#consuming-long-running-operations-using-operationt) |
+[Handling failures](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#reporting-errors-requestfailedexception) |
+[Diagnostics](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md) |
+[Mocking](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md#mocking) |
+[Client lifetime](https://devblogs.microsoft.com/azure-sdk/lifetime-management-and-thread-safety-guarantees-of-azure-sdk-net-clients/)
+<!-- CLIENT COMMON BAR -->
+
+## Examples
+
+You can familiarize yourself with different APIs using [Samples](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/maps/Azure.Maps.Search/tests/Samples).
+
+Before calling search APIs, instantiate a `SearchClient` first. Below use AAD to create the client instance:
+
+
+```C# Snippet:GetPolygons
+// Get Client
+var client = CreateClient();
+
+// Get Addresses
+var searchResult = await client.SearchAddressAsync("Seattle");
+
+// Extract geometry ids from addresses
+var geometry0Id = searchResult.Value.Results.First().DataSources.Geometry.Id;
+var geometry1Id = searchResult.Value.Results[1].DataSources.Geometry.Id;
```

Comment:
Can we get the coordinate here? geometry ID is less useful for users?

Reply:
The function getPolygons takes in a list of geometry ids as parameter; this demonstrates how users can extract the geometry id